persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -254,10 +254,21 @@ public static void inferRelationDetails(Module entityModule) {
                                         }
                                         // one-to-one
                                         if (!field.isArrayType() && !assocfield.isArrayType()) {
-                                            field.setRelation(computeRelation(entity, assocEntity, true,
-                                                    Relation.RelationType.ONE));
-                                            assocfield.setRelation(computeRelation(assocEntity, entity, false,
-                                                    Relation.RelationType.ONE));
+                                            if (!field.isOptionalType() && assocfield.isOptionalType()) {
+                                                field.setRelation(computeRelation(entity, assocEntity, true,
+                                                        Relation.RelationType.ONE));
+                                                assocfield.setRelation(computeRelation(assocEntity, entity, false,
+                                                        Relation.RelationType.ONE));
+                                            } else if (field.isOptionalType() && !assocfield.isOptionalType()) {
+                                                field.setRelation(computeRelation(entity, assocEntity, false,
+                                                        Relation.RelationType.ONE));
+                                                assocfield.setRelation(computeRelation(assocEntity, entity, true,
+                                                        Relation.RelationType.ONE));
+                                            } else {
```

Comment:
`if (!field.isOptionalType() && assocfield.isOptionalType())` is this condition covered here

Reply:
yes, it's covered in line 257 right?